### PR TITLE
adding authelia

### PR DESCRIPTION
**Pull request by leodotcloud**

### CODE REVIEW

File: proposed/authelia/latest/templates/authelia-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      default_method: single_factor
+    access_control:
+      default_policy: deny
+      any:
+        - domain: {{- printf " '*.%s'" .Values.authelia.config.domain }}
+          policy: deny
+      groups: {}
+      users: {}
+    session:
+      secret: {{ .Values.authelia.config.session.secret }}
```
Comment (guangbochen) [inline proposed/authelia/latest/templates/authelia-configmap.yaml]: questions.yml `#22`: `secret will be auto-generated if not specified` should either be changed to `required=true` or add auto generate `{{ randAlphaNum 10 | b64enc | quote }}` in here.

File: proposed/authelia/latest/questions.yml [hunk 2]
```diff
@@ -0,0 +1,128 @@
+categories:
+- Authentication
+questions:
+- variable: authelia.enableHealtcheck
+  default: false
+  description: "Enable kubernetes health check"
+  label: Enable health check
+  type: boolean
```
Comment (guangbochen) [inline proposed/authelia/latest/questions.yml]: probably is better to add a group for those ungrouped variables.

File: proposed/authelia/latest/app-readme.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Authelia
+
+[Authelia](https://www.authelia.com/) is an open-source authentication and authorization providing 2-factor authentication and single sign-on (SSO) for your applications. It acts as a companion of reverse proxies by handling authentication and authorization requests.
+
+This chart bootstraps a deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager. This application depends on [MongoDB](https://www.mongodb.com/) and [Redis](https://redis.io/), which can be installed as part of this catalog.
+
+### TODO:
+Move the redis and mongo out of deployments and use subcharts.
```
Comment (guangbochen) [inline proposed/authelia/latest/app-readme.md]: apart from the app-readme, it is better to add the `README.md` for the chart.

File: proposed/authelia/latest/templates/authelia-mongo-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        volumeMounts:
+          - name: data-volume
+            mountPath: /data/db
+          - name: config-volume
+            mountPath: /data/configdb
+        # TODO: Fix this
+      volumes:
+        - name: data-volume
+          hostPath:
+            path: /data/storage/mongo/data
```
Comment (guangbochen) [inline proposed/authelia/latest/templates/authelia-mongo-deployment.yaml]: enabling PV/PVc should be required for MongoDB, I think this is already on ur TODO list.

File: proposed/authelia/latest/templates/authelia-configmap.yaml (hunk 1 above)
Comment (leodotcloud) [reply proposed/authelia/latest/templates/authelia-configmap.yaml]: I prefer not to auto-generate in the helm chart due to the issue brought up here: https://github.com/helm/helm/issues/3053. I prefer UI generating a random value for me.

File: proposed/authelia/latest/questions.yml (hunk 2 above)
Comment (leodotcloud) [reply proposed/authelia/latest/questions.yml]: Ok let me think of something.

File: proposed/authelia/latest/templates/authelia-configmap.yaml (hunk 1 above)
Comment (leodotcloud) [reply proposed/authelia/latest/templates/authelia-configmap.yaml]: The session secret specified here is part of a config file and I don't have control over the way application specifies this secret. Will try to file an upstream issue.

**Pull request closed without merging**